src/pages/ShareCodePage.tsx

```diff
@@ -64,6 +87,21 @@ function ShareCodePage({report}: ShareCodePageProps) {
         ? `${urlWithTrailingSlash}${ROUTES.REPORT_WITH_ID.getRoute(report.reportID)}`
         : `${urlWithTrailingSlash}${ROUTES.PROFILE.getRoute(currentUserPersonalDetails.accountID ?? '-1')}`;
 
+    const logo = isReport ? getLogoForWorkspace(report, policy) : (UserUtils.getAvatarUrl(currentUserPersonalDetails?.avatar, currentUserPersonalDetails?.accountID) as ImageSourcePropType);
+
+    // Default logos (avatars) are SVG and they require some special logic to display correctly
+    let logoSVG: React.FC<SvgProps> | undefined;
+    let logoBackground: string | undefined;
+    let logoColor: string | undefined;
```

Comment:
Do we need these variables? Can we just directly assign the values and default to undefined?

Reply:
Well if policy is missing then it makes no sense to call `StyleUtils.getDefaultWorkspaceAvatarColor(policy.id)` to get the values for the colors. If I don't call it then I can't easily destructure the color values to assign them.

Also I chose this way because it gives me 100% confidence that all these variables will either be `undefined` at the same time or have values.

I'm sure there is a different way to express this code. I can rewrite this but please drop a suggestion if you can.

Reply:
Makes sense. Thanks for clarifying!